jerry-core/ecma/operations/ecma-jobqueue.h

```diff
@@ -0,0 +1,63 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ECMA_JOB_QUEUE_H
+#define ECMA_JOB_QUEUE_H
+
+#ifndef CONFIG_DISABLE_ES2015_PROMISE_BUILTIN
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmajobqueue ECMA Job Queue related routines
+ * @{
+ */
+
+typedef struct ecma_job_t ecma_job_t;
+typedef void (*ecma_job_handler_t) (ecma_job_t *);
+
+/**
+ * Description of general ecma job.
+ * Usually, it is used as the header of a specific job
+ */
+struct ecma_job_t
+{
+  ecma_job_handler_t handler; /**< the handler funtion for the job */
+};
```

Comment:
Then the callback in jerryport.h should be 
```
typedef void (*jerry_job_handler_t) (uintptr_t);

void jerry_port_jobqueue_enqueue (jerry_job_handler_t, uintptr_t);
```

and in every callback body, we need to cast the uintptr_t to some specific job_t (e.g. ecma_job_promise_reaction_job_t)

Am I right?